cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamFileJanitor.java

```diff
@@ -115,4 +120,12 @@ void clean(StreamConfig config, long currentTime) throws IOException {
   private boolean isPartitionDirector(Location location) throws IOException {
     return (location.isDirectory() && location.getName().indexOf('.') > 0);
   }
+
+  private boolean isStreamExists(Id.Stream streamId) throws IOException {
```

Comment:
It's a legacy issue. The `StreamAdmin` used to extend from `DatasetAdmin`, which throws `Exception` in all methods. When we detach `StreamAdmin` from `DatasetAdmin`, those method signature get retained.